hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/VertexAPI.java

```diff
@@ -259,22 +306,58 @@ private static void checkBatchSize(HugeConfig config,
         int max = config.get(ServerOptions.MAX_VERTICES_PER_BATCH);
         if (vertices.size() > max) {
             throw new IllegalArgumentException(String.format(
-                      "Too many counts of vertices for one time post, " +
+                      "Too many vertices for one time post, " +
                       "the maximum number is '%s'", max));
         }
     }
 
-    @JsonIgnoreProperties(value = {"type"})
-    private static class JsonVertex implements Checkable {
+    private Id getVertexId(HugeGraph g, JsonVertex vertex) {
+        VertexLabel vertexLabel = g.vertexLabel(vertex.label);
+        String labelId = vertexLabel.id().asString();
+        IdStrategy idStrategy = vertexLabel.idStrategy();
+        E.checkArgument(idStrategy != IdStrategy.AUTOMATIC,
+                        "Automatic Id strategy is not supported now");
+
+        if (idStrategy == IdStrategy.PRIMARY_KEY) {
+            List<Id> pkIds = vertexLabel.primaryKeys();
+            List<Object> pkValues = new ArrayList<>(pkIds.size());
+            for (Id pkId : pkIds) {
+                String propertyKey = g.propertyKey(pkId).name();
+                Object propertyValue = vertex.properties.get(propertyKey);
+                E.checkState(propertyValue != null,
+                             "The value of primary key '%s' can't be null",
+                             propertyKey);
+                pkValues.add(propertyValue);
+            }
+
+            String value = SplicingIdGenerator.concatValues(pkValues);
+            return SplicingIdGenerator.splicing(labelId, value);
+        } else {
+            assert idStrategy == IdStrategy.CUSTOMIZE_NUMBER ||
+                   idStrategy == IdStrategy.CUSTOMIZE_STRING;
+            return HugeVertex.getIdValue(vertex.id);
+        }
+    }
+
+    private static class BatchVertexRequest {
 
-        @JsonProperty("id")
-        public Object id;
-        @JsonProperty("label")
-        public String label;
-        @JsonProperty("properties")
-        public Map<String, Object> properties;
-        @JsonProperty("type")
-        public String type;
+        @JsonProperty("vertices")
+        public List<JsonVertex> jsonVertices;
+        @JsonProperty("update_strategies")
+        public Map<String, UpdateStrategy> updateStrategies;
+        @JsonProperty("create_if_not_exist")
+        public boolean createIfNotExist = true;
```

Comment:
we don't seem to deal with situation the user passes parameter create_if_not_exist=false

Reply:
maybe we can just throw NotsupportException if create_if_not_exist=false